### ISSUE TITLE
ExplicitItLambdaParameter rule is not working

### ISSUE THREAD

**Issue by adamkobor** [detekt/detekt#3245]
Hey there, I'm experiencing a weird bug with the `ExplicitItLambdaParameter` rule. I've enabled it in my config.yml, but it definitely doesn't trigger any rule violation.
 
## Expected Behavior
It should trigger a rule violation for multi-line lambdas with implicit `it` parameters.

## Observed Behavior
It doesn't trigger a rule violation in the cases described above.

## Steps to Reproduce
Write a multi-line lambda with an implicit `it` parameter, and run detekt:

```kotlin
listOf("a","b").forEach {
    doSomething()
    println(it)
}
```

## Your Environment

* Version of detekt used: 1.14.2
* Version of Gradle used (if applicable): 6.7.1
* Operating System and version: MacOS Big Sur

**Comment by cortinico** [comment]
As described in the rule documentation: https://detekt.github.io/detekt/style.html#explicititlambdaparameter
your code is actually valid, as you're using an **implicit** `it` parameter. The rule flags **explicit** usages of the `it` parameter.

**Comment by adamkobor** [comment]
OMG, what a rookie mistake (and also a misread) 😄 Thanks, I was concentrating only to the word "lambda" in the rule's name

**Comment by QuinnBast** [comment]
Hello. Just stumbled across this. Is there a rule that forces lambda parameters to be defined?

**Comment by BraisGabin** [comment]
Your request is interesting but it's not related with this issue. Could you open a new issue so we can talk about it?

(Github should allow to move one comment directly to a new issue)
